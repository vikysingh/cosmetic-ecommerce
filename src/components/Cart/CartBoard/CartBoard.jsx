import React from "react"

import { Typography } from "antd"

import globalStyles from "../../../styles/Globals.module.css"
import styles from "./CartBoard.module.css"
 
import { ThemeButton } from "../../index"
import PropTypes from 'prop-types'

import { Link } from "react-router-dom"

import routes from "../../../constants/routes.json"

import { useHistory } from "react-router-dom"

import auth from "../../../utils/auth"

const { Title } = Typography

function CartBoard({ subtotal, shippingPrice, total }) {

    let history = useHistory()

    return <div className={styles.CART_BOARD} >
        <Title level={3}> Order Summary </Title>
        <hr />
        <div className={globalStyles.flexRowBetweenCenter} >
            <Title level={4}> Subtotal </Title>
            <strong> €{subtotal} </strong> 
        </div>
        <div className={globalStyles.flexRowBetweenCenter} >
            <Title level={4}> Shipping </Title>
            <strong> €{shippingPrice} </strong> 
        </div>
        <hr />
        <div className={globalStyles.flexRowBetweenCenter} >
            <Title level={4}> Total </Title>
            <strong> €{total} </strong> 
        </div>
        <button onClick={() => {
            auth.allowCheckout(() => {
                history.push(routes.checkoutAddress)
            })
        }} > 
            {/* <Link to={routes.checkoutAddress} > */}
                Go to checkout
            {/* </Link> */}
        </button>
    </div>
}

CartBoard.propTypes = {
    subtotal: PropTypes.number.isRequired,
    shippingPrice: PropTypes.number.isRequired,
    total: PropTypes.number.isRequired
}


export default React.memo(CartBoard)