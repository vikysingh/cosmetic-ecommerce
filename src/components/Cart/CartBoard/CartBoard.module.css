.CART_BOARD {
    width: 90%;
    background-color: #2f3640;
    color: #fff;
    padding: 30px 15px;
    border-radius: 5px;
    margin: auto;
}

.CART_BOARD h3 {
    color: #fff;
}

.CART_BOARD > hr {
    margin: 20px 0;
    color: #fff;
}

.CART_BOARD > hr:nth-child(2) {
    margin-top: 40px;
}

.CART_BOARD h4 {
    font-size: 0.9rem;
    margin: 5px 0;
    color: #fff;
}

.CART_BOARD button {
    width: 100%;
    margin-top: 40px;
    height: 50px;
    font-weight: bolder;
    background-color: #1B9CFC;
    border: none;
}

.CART_BOARD button:hover {
    color: #1B9CFC;
}