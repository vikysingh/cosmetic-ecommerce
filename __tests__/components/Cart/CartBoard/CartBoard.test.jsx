import React from "react"
import {  shallow } from "enzyme"

import CartBoard from "../../../../components/Cart/CartBoard/CartBoard"
import { dummyCartBoardData } from "../../../../test_files/dummyTestData"

test('Should render CartBoard from Cart', () => {
    const wrapper = shallow(<CartBoard subtotal={dummyCartBoardData.subrotal}
        shippingPrice={dummyCartBoardData.shippingPrice} total={dummyCartBoardData.totla} />)

    expect(wrapper).toMatchSnapshot()
})