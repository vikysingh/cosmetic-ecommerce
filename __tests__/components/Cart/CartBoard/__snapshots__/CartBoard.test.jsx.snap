// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CartBoard from Cart 1`] = `
<div
  className="CART_BOARD"
>
  <Title
    level={3}
  >
     Order Summary 
  </Title>
  <hr />
  <div
    className="flexRowBetweenCenter"
  >
    <Title
      level={4}
    >
       Subtotal 
    </Title>
    <strong>
       €
      20
       
    </strong>
  </div>
  <div
    className="flexRowBetweenCenter"
  >
    <Title
      level={4}
    >
       Shipping 
    </Title>
    <strong>
       €
      5
       
    </strong>
  </div>
  <hr />
  <div
    className="flexRowBetweenCenter"
  >
    <Title
      level={4}
    >
       Total 
    </Title>
    <strong>
       €
       
    </strong>
  </div>
  <[object Object]
    theme="shop"
  >
     
    <[object Object]
      to="/checkout/address"
    >
      Go to checkout
    </[object Object]>
     
  </[object Object]>
</div>
`;
